lib/colmena/error.rb

```diff
@@ -1,10 +1,16 @@
+# frozen_string_literal: true
+
 module Colmena
   module Error
-    def error(name, data = {})
+    def self.new(name, data = {})
       {
         type: name,
         data: data,
       }
     end
```

Comment:
Maybe this could be misleading? In Ruby, when you use `Something.new` you expect an instance of `Something`. We could rename it to `call`, then we could use `Colmena::Error(:some_error)`.

Reply:
I think you're right that it can be misleading, but the alternative you propose doesn't work AFAIK.

The accepted syntax is `Error.(:some_error)`, which I guess is what you meant.

For the record, I started with `self.call` and that syntax, but Rubocop told me that syntax is discouraged.  See https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/LambdaCall

So I pivoted to `Error.new` because I thought it was clearer than `Error.call`.

I'm not married to either, though. We could also use `Error.error`, `Error.structure`, ...